hydradx/tests/test_omnipool_amm.py

```diff
@@ -2491,3 +2491,139 @@ def test_buy_sell_spot(hdx_lrna: float, usd_lrna: float, asset_fee: float, lrna_
+    if lrna_price != pytest.approx(usd_amt / usd_lrna, rel=1e-15):
+        raise AssertionError(f'lrna_price {lrna_price} != {usd_amt / usd_lrna}')
+    if usd_price != pytest.approx(usd_lrna / usd_amt, rel=1e-15):
+        raise AssertionError(f'lrna_price {usd_price} != {usd_lrna / usd_amt}')
+
+
+@given(st.lists(asset_quantity_strategy, min_size=6, max_size=6),
+       st.floats(min_value=0.0001, max_value=0.1, exclude_min=True))
+def test_sell_spot_LRNA(amts: list, asset_fee: float):
+    '''Tests sell_spot with LRNA as the sell_tkn'''
+
+    hdx_amt, usd_amt, dot_amt = mpf(amts[0]), mpf(amts[1]), mpf(amts[2])
+    hdx_lrna, usd_lrna, dot_lrna = mpf(amts[3]), mpf(amts[4]), mpf(amts[5])
+
+    initial_state = oamm.OmnipoolState(
+        tokens={
+            'HDX': {'liquidity': hdx_amt, 'LRNA': hdx_lrna},
+            'USD': {'liquidity': usd_amt, 'LRNA': usd_lrna},
+            'DOT': {'liquidity': dot_amt, 'LRNA': dot_lrna},
+        },
+        lrna_fee=0.0005,
+        asset_fee=asset_fee,
+        preferred_stablecoin='USD'
+    )
+
+    price = initial_state.sell_spot('LRNA', 'USD')
+    if price != pytest.approx(usd_amt / usd_lrna * (1-asset_fee), rel=1e-15):
+        raise AssertionError(f'price {price} is incorrect')
+
+
+@given(st.lists(asset_quantity_strategy, min_size=6, max_size=6),
+       st.floats(min_value=0.0001, max_value=0.1, exclude_min=True))
+def test_buy_spot_LRNA(amts: list, asset_fee: float):
+    '''Tests buy_spot with LRNA as the sell_tkn'''
+
+    hdx_amt, usd_amt, dot_amt = mpf(amts[0]), mpf(amts[1]), mpf(amts[2])
+    hdx_lrna, usd_lrna, dot_lrna = mpf(amts[3]), mpf(amts[4]), mpf(amts[5])
+
+    initial_state = oamm.OmnipoolState(
+        tokens={
+            'HDX': {'liquidity': hdx_amt, 'LRNA': hdx_lrna},
+            'USD': {'liquidity': usd_amt, 'LRNA': usd_lrna},
+            'DOT': {'liquidity': dot_amt, 'LRNA': dot_lrna},
+        },
+        lrna_fee=0.0005,
+        asset_fee=asset_fee,
+        preferred_stablecoin='USD'
+    )
+
+    price = initial_state.buy_spot('USD', 'LRNA')
+    exp_price = usd_lrna / usd_amt
+    if price != pytest.approx(exp_price/(1-asset_fee), rel=1e-15):
+        raise AssertionError(f'price {price} is incorrect')
+
+
+def test_value_assets_without_equivalency_map():
+    initial_state = oamm.OmnipoolState(
+        tokens={
+            'HDX': {'liquidity': mpf(1000000000), 'LRNA': mpf(100000000)},
+            'USD': {'liquidity': mpf(1000000000), 'LRNA': mpf(1000000000)},
+            'DOT': {'liquidity': mpf(1000000000), 'LRNA': mpf(10000000000)},
+        },
+        lrna_fee=0.0025,
+        asset_fee=0.0005,
+        preferred_stablecoin='USD'
+    )
+
+    assets = {'HDX': mpf(1000), 'USD': mpf(2000), 'DOT': mpf(3000), 'LRNA': mpf(4000)}
+    val = initial_state.value_assets(assets, stablecoin='USD')
+    if val != 100 + 2000 + 30000 + 4000:
+        raise AssertionError(f'val {val} is incorrect')
+
+
+def test_no_preferred_stablecoin():
```

Comment:
One thing that still doesn't work with no stablecoin is repr(). It doesn't break anything critical, and I added a fix for that in my latest branch, so I think it's ok if we merge this as is.

Reply:
It works, I had already commented out the line with the stablecoin reference. I added it to the test for now. Your solution will be better than just commenting out that line I am guessing, but this works for now.